src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    if (this.health > 0) {
+      Animal.alive.push(this);
+    }
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore && prey.hidden === false) {
+      prey.health -= 50;
+
+      if (prey.health === 0) {
```

Comment:
The condition `prey.health === 0` should be updated to `prey.health <= 0` to ensure that any prey with health dropping below zero is correctly identified as dead. This is crucial for handling cases where multiple bites might reduce health below zero.